cli/cppcheckexecutor.cpp

```diff
@@ -194,7 +194,7 @@ int CppCheckExecutor::check(int argc, const char* const argv[])
     mStdLogger = new StdLogger(settings);
 
     CppCheck cppCheck(*mStdLogger, true, executeCommand);
-    cppCheck.settings() = settings;
+    cppCheck.settings() = std::move(settings);
```

Comment:
This is a bogus warning. The object is referenced in `StdLogger`.

Reply:
The `Settings` object should also be `const` after construction to make sure it is no longer modified afterwards. That is still WIP.

Also the copy will go away after #4964 is done.

Reply:
This is a variant of https://trac.cppcheck.net/ticket/12364.